libnavigation-core/src/androidTest/java/com/mapbox/navigation/core/trip/service/ArtificialDriverTest.kt

```diff
@@ -0,0 +1,146 @@
+                    mapboxNavigation.historyRecorder.stopRecording {
+                        continuation.resume(it ?: "null")
+                    }
+                }
+                val offRouteState = states.filter { it.routeState == RouteState.OFF_ROUTE }
+                assertTrue(
+                    "${offRouteState.size} off-route states have been detected(" +
+                        "more info in $historyFile): $offRouteState",
+                    offRouteState.isEmpty()
+                )
+            }
+        }
+}
+
+private fun createArtificialLocationUpdates(
+    testRoute: NavigationRoute
+): List<ReplayEventUpdateLocation> {
+    val replayRouteMapper = ReplayRouteMapper()
+    return replayRouteMapper
+        .mapDirectionsRouteGeometry(testRoute.directionsRoute)
+        .filterIsInstance<ReplayEventUpdateLocation>()
+}
+
+private suspend fun MapboxNativeNavigator.collectStatuses(
+    untilRouteState: RouteState
+): MutableList<NavigationStatus> {
+    val statues = mutableListOf<NavigationStatus>()
+    statusUpdates()
+        .map { it.status }
+        .takeWhile { it.routeState != untilRouteState }
+        .toList(statues)
+    return statues
+}
+
+data class OnStatusUpdateParameters(
+    val origin: NavigationStatusOrigin,
+    val status: NavigationStatus
+)
+
+@OptIn(ExperimentalCoroutinesApi::class)
+fun MapboxNativeNavigator.statusUpdates(): Flow<OnStatusUpdateParameters> {
+    return callbackFlow {
+        val observer = NavigatorObserver { origin, status ->
+            this.trySend(OnStatusUpdateParameters(origin, status))
+        }
+        addNavigatorObserver(observer)
+        awaitClose {
+            removeNavigatorObserver(observer)
+        }
+    }
+}
+
+private suspend fun withNavigators(
+    block: suspend (MapboxNavigation, MapboxNativeNavigator) -> Unit
+) {
+    val context = InstrumentationRegistry.getInstrumentation().targetContext
+    val mapboxNavigation = MapboxNavigationProvider.create(
+        NavigationOptions.Builder(context)
+            .accessToken(context.getString(R.string.mapbox_access_token))
+            .build()
+    )
+    try {
+        block(mapboxNavigation, MapboxNativeNavigatorImpl)
+    } finally {
+        mapboxNavigation.onDestroy()
+    }
+}
+
+private fun getTestRoute(): NavigationRoute {
+    val context = InstrumentationRegistry.getInstrumentation().targetContext
+    return NavigationRoute.create(
+        directionsResponseJson = context.resources.openRawResource(R.raw.test_long_route)
+            .readBytes().decodeToString(),
+        routeRequestUrl = "https://api.mapbox.com/directions/v5/mapbox/driving/" +
+            "11.566744%2C48.143769%3B8.675521%2C50.119087" +
+            "?alternatives=false" +
+            "&geometries=polyline6" +
+            "&language=en" +
+            "&overview=full" +
+            "&steps=true" +
+            "&access_token=YOUR_MAPBOX_ACCESS_TOKEN",
```

Comment:
See `NativeNavigatorCallbackOrderTest`, it already uses access token that is being injected on CI.

Reply:
Ah, you don't want it to be present in the URL.